Avatar rating text is not translated properly if a user has a custom language Set

ActuallyConnor opened the pull request:
### Description of the Change

Adding string translation method around each `$avatar_rating` when it is output in the `edit_user_profile()` method.

### Verification Process

- [x] Site language is set to English

![Screen Shot 2021-11-17 at 9 29 43 AM](https://user-images.githubusercontent.com/20844837/142219599-d72beb31-288d-4ee2-af54-688c5ae538f0.png)

- [x] User language is set to French

![Screen Shot 2021-11-17 at 9 29 37 AM](https://user-images.githubusercontent.com/20844837/142219739-c909410a-65fc-4001-a544-948e5f9dd915.png)

- [x] Rating strings are translated to French

![Screen Shot 2021-11-17 at 9 29 28 AM](https://user-images.githubusercontent.com/20844837/142219795-1db4ad69-3c5a-4607-b0f9-9fe63b17bf9c.png)


### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

<!-- _NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open._ -->

### Applicable Issues

[https://github.com/10up/simple-local-avatars/issues/88](https://github.com/10up/simple-local-avatars/issues/88)

ActuallyConnor committed:
Avatar rating text is not translated properly if a user has a custom language set

faisal-alvi commented on includes/class-simple-local-avatars.php:
```diff
@@ -407,7 +407,7 @@ class="button item-delete submitdelete deletion"
 
 					foreach ( $this->avatar_ratings as $key => $rating ) :
-						echo "\n\t<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( $key ) . "' " . checked( $profileuser->simple_local_avatar_rating, $key, false ) . "/> $rating</label><br />";
+						echo "\n\t<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( $key ) . "' " . checked( $profileuser->simple_local_avatar_rating, $key, false ) . "/>" . __($rating) . "</label><br />";
```
@ActuallyConnor Thank you for raising a PR!

This looks good and fixes the issue but as per standards, the plain strings should always be directly passed as a string literal in the translation function, not as a variable. Read more [here](https://developer.wordpress.org/reference/functions/__/#comment-540).

> This seems to work, but it will interfere in automatic parsing of your plugin/theme’s files for translation.

Moreover, we are already translating strings.

https://github.com/10up/simple-local-avatars/blob/86ce042ae571583986d0c439675fc3f9be5be04a/includes/class-simple-local-avatars.php#L15-L20

But these are loaded prior to the user object being initialized as mentioned in #88. That needs to be fixed in a standard way.

ActuallyConnor replied:
Understood, I can work on fixing that to work in a standard way

ActuallyConnor committed:
Outputting each rating individually here

ActuallyConnor commented:
@faisal-alvi I implemented a new solution and left some comments for clarity. Let me know if you have any questions or concerns.

ActuallyConnor committed:
Fixing indentation

ActuallyConnor committed:
Fixing indentation

ActuallyConnor committed:
Fixing indentation

ActuallyConnor committed:
Fixing indentation

faisal-alvi commented on includes/class-simple-local-avatars.php:
```diff
@@ -406,9 +406,12 @@ class="button item-delete submitdelete deletion"
-					foreach ( $this->avatar_ratings as $key => $rating ) :
-						echo "\n\t<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( $key ) . "' " . checked( $profileuser->simple_local_avatar_rating, $key, false ) . "/> $rating</label><br />";
-					endforeach;
+                                        // These are the same as $this->avatar_ratings but in order for string translations to appear we have to output them like the following
+                                        // We also use the default text domain here
+                                        echo "<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( 'G' ) . "' " . checked( $profileuser->simple_local_avatar_rating, 'G', false ) . "/>" . __( 'G &#8212; Suitable for all audiences' ) . "</label><br />";
+                                        echo "<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( 'PG' ) . "' " . checked( $profileuser->simple_local_avatar_rating, 'PG', false ) . "/>" . __( 'PG &#8212; Possibly offensive, usually for audiences 13 and above' ) . "</label><br />";
+                                        echo "<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( 'R' ) . "' " . checked( $profileuser->simple_local_avatar_rating, 'R', false ) . "/>" . __( 'R &#8212; Intended for adult audiences above 17' ) . "</label><br />";
+                                        echo "<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( 'X' ) . "' " . checked( $profileuser->simple_local_avatar_rating, 'X', false ) . "/>" . __( 'X &#8212; Even more mature than above' ) . "</label><br />";
```
Moving it to a separate function and updating `$this->avatar_ratings`, and calling just before the `foreach` starts - would be a better improvement. Then no need to change anything in the `foreach` at all.

The function would just have the same array from `__constructor`, overwriting the existing `$this->avatar_ratings`.

https://github.com/10up/simple-local-avatars/blob/86ce042ae571583986d0c439675fc3f9be5be04a/includes/class-simple-local-avatars.php#L15-L20

Also, using `esc_html` over `$ratings` would be an extra improvement.

https://github.com/10up/simple-local-avatars/blob/86ce042ae571583986d0c439675fc3f9be5be04a/includes/class-simple-local-avatars.php#L410

faisal-alvi commented:
@ActuallyConnor Thank you for the improved solution. I've left some suggestions, please check.

ActuallyConnor committed:
Changes based on code review

Overwriting existing avatar_ratings so this can be called just before the rating strings would be used so that translations will work correctly.

Default text-domain because the strings have already been translated

Using esc_html over $ratings would be an extra improvement

ActuallyConnor committed:
Changing to updateAvatarRatings and moving call higher in the code

ActuallyConnor commented:
@faisal-alvi I've made some changes based on your recommendations, let me know what you think

faisal-alvi commented on includes/class-simple-local-avatars.php:
```diff
@@ -401,15 +401,19 @@ class="button item-delete submitdelete deletion"
-						echo "\n\t<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( $key ) . "' " . checked( $profileuser->simple_local_avatar_rating, $key, false ) . "/> $rating</label><br />";
-					endforeach;
-					?>
+						if ( empty( $profileuser->simple_local_avatar_rating ) || ! array_key_exists( $profileuser->simple_local_avatar_rating,
+								$this->avatar_ratings ) ) {
```
407 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
408 | ERROR | [x] Closing parenthesis of a multi-line function call must be on a line by itself

faisal-alvi commented on includes/class-simple-local-avatars.php:
```diff
@@ -401,15 +401,19 @@ class="button item-delete submitdelete deletion"
+						}
+
+						foreach ( $this->avatar_ratings as $key => $rating ) :
+							echo "\n\t<label><input type='radio' name='simple_local_avatar_rating' value='" . esc_attr( $key ) . "' " . checked( $profileuser->simple_local_avatar_rating,
+									$key, false ) . "/>" . esc_html( $rating ) . "</label><br />";
```
413 | ERROR | [x] Opening parenthesis of a multi-line function call must be the last content on the line
414 | ERROR | [x] Closing parenthesis of a multi-line function call must be on a line by itself

faisal-alvi commented on includes/class-simple-local-avatars.php:
```diff
@@ -678,4 +682,18 @@ public function get_avatar_rest( $user ) {
 	public function set_avatar_rest( $input, $user ) {
 		$this->assign_new_user_avatar( $input['media_id'], $user->ID );
 	}
+
+    /**
+     * Overwriting existing avatar_ratings so this can be called just before the rating strings would be used so that
+     * translations will work correctly.
+     * Default text-domain because the strings have already been translated
+     */
+    private function update_avatar_ratings() {
+        $this->avatar_ratings = array(
+            'G'  => __( 'G &#8212; Suitable for all audiences' ),
+            'PG' => __( 'PG &#8212; Possibly offensive, usually for audiences 13 and above' ),
+            'R'  => __( 'R &#8212; Intended for adult audiences above 17' ),
+            'X'  => __( 'X &#8212; Even more mature than above' ),
+        );
+    }
```
684-698 | ERROR | [x] Tabs must be used to indent lines; spaces are not allowed

faisal-alvi reviewed (changes requested):
@ActuallyConnor the solution looks good, thank you. There are a couple of errors from `PHPCS`, please fix them. Then we will be good to go.

ActuallyConnor committed:
Fixing phpcs errors

ActuallyConnor commented:
@faisal-alvi those PHPCS errors should be fixed now 😄
